src/Core/Math/LinearAlgebra.hpp

```diff
@@ -156,15 +156,15 @@ namespace Ra
             inline bool checkRange( const Vector_& v, const Scalar& min, const Scalar& max );
 
             /// Get two vectors orthogonal to a given vector.
-            inline void getOrthogonalVectors( const Vector3& fx, Vector3& fy, Vector3& fz );
+            inline void getOrthogonalVectors( const Vector3& fx, Eigen::Ref<Vector3> fy, Eigen::Ref<Vector3> fz );
 
             /// Get the angle between two vectors. Works for types where the cross product is
             /// defined (i.e. 2D and 3D vectors).
             template<typename Vector_>
             inline Scalar angle( const Vector_& v1, const Vector_& v2 );
 
             /// @return the projection of point on the plane define by plane and planeNormal
-            inline Vector3 projectOnPlane(const Vector3& planePos, const Vector3 planeNormal, const Vector3& point);
+            inline Vector3 projectOnPlane(const Vector3& planePos, const Vector3& planeNormal, const Vector3& point);
```

Comment:
> For non-const refs, we must use Eigen::Ref if we ever want to use something like a matrix column as an argument, otherwise the compiler refuses it.

I do agree, Eigen::Ref is the proper solution for that.

> In the case of const ref the compiler accept it but it will generate a copy (or it could optimize it).
> The question is, do we want to live with this copy in this specific use case (which is very rare) 

As stated by Eigen guys:
> 18:47 <@ChriSopht> even copying a 100d vector does not cost that much. as always avoid premature optimization

having a copy (especially on such rare case, as  you said) is not critical, and might be optimized by the compiler (as you said). So, I would recommend to use const Ref, to have consistent API between const and non-const functions, and simpler coding rules. In general, the overhead should be almost 0.

> or modify all our code accepting const Vector3&

I'm not sure these practical refactoring consideration should influence or code design.